src/merkletree.js

```diff
@@ -161,7 +163,10 @@ class MerkleTree {
         }
         const path = []
         for (let i = index; i > 1; i >>= 1) {
-            path.push(hashes[i ^ 1])  // the other sibling
+            const otherSibling = hashes[i ^ 1]
+            if (otherSibling.compare(ZERO) !== 0) {
```

Comment:
@jtakalai IIUC might be a bit clearer if you can use `Buffer#equals` instead of `Buffer#compare`?

i.e.
```js
if (!otherSibling.equals(ZERO)) {
    path.push(otherSibling)
}
```
Just a thought.

Reply:
changed that one in https://github.com/streamr-dev/monoplasma/pull/21